### ISSUE TITLE
Permits for users

### ISSUE THREAD

**Issue by scepeda78** [agentlabs-inc/agentlabs#45]
Lets say we have agents that can be accesed for a group of users (for example  an agent that handles sensitivo information of a company),then most probably a system of permits is needed for the users.

**Comment by kevinpiac** [comment]
Interesting! 

So if I understand, you would like for example to restrict an Agent to a specific group of users?

**Comment by scepeda78** [comment]
Yes, because that agent for example is connected to sensitive data of the company (balance sheet, or projects being done or agreements with clienta, as example)



El 21-10-2023 6:11 a. m., Kevin Piacentini ***@***.***> escribió:

Interesting!

So if I understand, you would like for example to restrict an Agent to a specific group of users?

**Comment by kevinpiac** [comment]
It makes sense. I cannot garantee when it will ship, but it's definitely something we'll consider.

Thank you much for your suggestions 🙏

**Comment by kevinpiac** [comment]
Just adding some more information here.

In the next release, we'll allow you to authenticate users directly from the SDK.

It will work as follow:

```typescript
agent.requestLogin({ text: 'Please login to access this agent' })
```

And this will send a rich component into the chat to ask the user to authenticate (see screenshot).

<img width="1110" alt="Screenshot 2023-10-21 at 13 44 48" src="https://github.com/agentlabs-inc/agentlabs/assets/15961122/bcbe00d5-88a6-4e5d-bde6-8d5277ef9aa3">

We can imagine the same king of control for groups.
You could add a user to a group.
Then from the SDK you would just do something as follows:

```typescript
project.onUserMessage(() => {
  if (user.group === 'some-group') {
     agentA.send('Agent A is available')
   }
   
   if (user.group === 'some-other-group') {
      agentB.send('You have access to this VIP agent')
   }
});
```

Would this make sense to you?

**Comment by scepeda78** [comment]
To be honest, I'm not expert in coding! Because meanwhile I ve think in other option, which is to have a separated database for that agent, not maybe the best way but it might work.

El 21-10-2023 8:47 a. m., Kevin Piacentini ***@***.***> escribió:

Just adding some more information here.

In the next release, we'll allow you to authenticate users directly from the SDK.

It will work as follow:

agent.requestLogin({ text: 'Please login to access this agent' })

And this will send a rich component into the chat to ask the user to authenticate (see screenshot).

[Screenshot 2023-10-21 at 13 44 48]<https://user-images.githubusercontent.com/15961122/277106072-bcbe00d5-88a6-4e5d-bde6-8d5277ef9aa3.png>

We can imagine the same king of control for groups.
You could add a user to a group.
Then from the SDK you would just do something as follows:

project.onUserMessage(() => {
  if (user.group === 'some-group') {
     agentA.send('Agent A is available')
   }

   if (user.group === 'some-other-group') {
      agentB.send('You have access to this VIP agent')
   }
});

Would this make sense to you?
